src/core/handle.ml

```diff
@@ -186,7 +188,48 @@ let handle_cmd : sig_state -> p_command -> sig_state * proof_data option =
       out 3 "(symb) %s ≔ %a\n" x.elt pp_term t;
       let d = if op then None else Some(t) in
       let ss = Sig_state.add_symbol ss e Defin x a impl d in
-      (ss, None)
+      (ss, None, None)
+  | P_inductive(e, s, t, tl) ->
+      (* Obtaining of the global type *)
+      let type_symbol_head = match t with
+        | None -> { elt = P_Type ; pos = None }
+        | Some a -> a
+      in
+      (* Add the first symbol in the signature *)
+      let (ss, _, _) = handle_cmd ss
+                      ({elt = P_symbol(e, Defin, s, [], type_symbol_head);
+                        pos = cmd.pos})
+      in
+      let create_symbol : string -> term -> term option -> sym  =
+        fun s t def ->
+        { sym_name = s ; sym_type = ref t ;
+          sym_path = file_to_module (current_path ());
+          sym_def = ref def ; sym_impl  = [] ; sym_rules = ref [];
+          sym_tree  = ref Tree_types.empty_dtree ;
+          sym_prop = Injec ; sym_expo = e }
+      in
+      (* Add recursively each constructor in signature
+         and create the sym list                    *)
+      let rec add_constructor : sig_state -> sym list -> (ident * p_term) list
+                                -> sig_state * sym list =
+        fun ss_cur syml l -> match l with
+                   | [] -> (ss_cur, syml)
+                   | (ident, pterm)::q ->
+                       let (ss_new, _, _) =
+                         handle_cmd ss_cur
+                           ({elt = P_symbol(e, Injec, ident, [], pterm);
+                             pos = cmd.pos})
+                       in
+                       let term = scope_term e ss_cur Env.empty pterm in
```

Comment:
Scoping is done in handle_symbol.

Reply:
Yes but I need again this information when I would like to build my record.
I can change the type of my function `handle_symbol` to obtain a pair with the scoping term, but I'm not sure it's better.